OWML.ModHelper.Input/ModInputCombination.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using OWML.Common;
+using UnityEngine;
+
+namespace OWML.ModHelper.Input
+{
+    public class ModInputCombination : IModInputCombination
+    {
+        private const int MaxUsefulKey = 350;
+        private const int MaxComboLength = 7;
+        private const string XboxPrefix = "xbox_";
+
+        public float LastPressedMoment { get; private set; }
+        public bool IsFirst { get; private set; }
+        public float PressDuration => LastPressedMoment - _firstPressedMoment;
+        public string ModName { get; }
+        public string Name { get; }
+        public ReadOnlyCollection<KeyCode> Singles => _singles.AsReadOnly();
+        public ReadOnlyCollection<long> Hashes => _hashes.AsReadOnly();
+
+        private bool _isPressed;
+        private float _firstPressedMoment;
+        private List<KeyCode> _singles = new List<KeyCode>();
+        private List<long> _hashes = new List<long>();
+
+        private KeyCode StringToKeyCodeKeyboard(string keyboardKey)
+        {
+            if (keyboardKey == "control" || keyboardKey == "ctrl")
+            {
+                return KeyCode.LeftControl;
+            }
+            if (keyboardKey == "shift")
+            {
+                return KeyCode.LeftShift;
+            }
+            if (keyboardKey == "alt")
+            {
+                return KeyCode.LeftAlt;
+            }
+            var code = (KeyCode)Enum.Parse(typeof(KeyCode), keyboardKey, true);
+            return Enum.IsDefined(typeof(KeyCode), code) ? code : KeyCode.None;
+        }
+
+        private KeyCode StringToKeyCodeXbox(string xboxKey)
+        {
+            var xboxCode = (XboxButton)Enum.Parse(typeof(XboxButton), xboxKey, true);
+            return (Enum.IsDefined(typeof(XboxButton), xboxCode)) ?
+                InputTranslator.GetKeyCode(xboxCode, false) : KeyCode.None;
+        }
+
+        private KeyCode StringToKeyCode(string key)
+        {
+            key = key.Trim();
```

Comment:
instead of overwriting the variable it's nicer to create a new one: `var trimmedKey`, then use that. Not a big deal in this example, but it's a good habit. Makes it easier to find the state of a system :)

Reply:
still not? :P

Reply:
do I have to? 😋

Reply:
i guess not 🤷